### PR TITLE
pytest: Change pytest tests function signature - Option 2

### PR DESCRIPTION
**Pull request by marioevz**
Based on suggestion by @spencer-tb, this PR implements another option for the signature of pytest based test functions.

It implements an filled-at-instantiation class that is defined right before the test function is called, and is tailored to the specific test function.

### CODE REVIEW

**Comment by danceratopz** [comment]
I really like it. I think it's exactly the improvement in test writing ux we were looking for!

File: conftest.py [hunk 1]
```diff
@@ -191,35 +196,57 @@ def pytest_runtest_call(item):
...
+                    reference_spec,
+                    eips=eips,
+                )
+                write_fixture_file(item, fixture_output)
+
+        item.funcargs["state_test"] = StateTestWrapper
+
+    if "blockchain_test" in item.funcargs:
+
+        class BlockchainTestWrapper(StateTest):
```
Comment (danceratopz) [inline conftest.py:227]: Minor typo, the wrapper should derive from `BlockchainTest` not `StateTest` here.

**Comment by danceratopz** [comment]
I love this solution!

File: conftest.py (hunk 1 above)
Comment (danceratopz) [reply conftest.py:227]: Will merge to the main pytest branch and fix there!

**Comment by danceratopz** [comment]
Moved the wrapper class definitions to the fixture functions (for discussion) in https://github.com/marioevz/execution-spec-tests/pull/4

File: conftest.py (hunk 1 above)
Comment (danceratopz) [reply conftest.py:227]: This is fixed via https://github.com/marioevz/execution-spec-tests/pull/4.

**Comment by marioevz** [comment]
Superseded by https://github.com/danceratopz/execution-spec-tests/pull/4

**Pull request closed without merging**